Cannot build samples/net/sockets/echo for cc3220sf_launchxl

**Describe the bug**
Cannot build samples/net/sockets/echo for cc3220sf_launchxl

**To Reproduce**
Steps to reproduce the behavior:
1. build samples/net/sockets/echo for cc3220sf_launchxl

**Expected behavior**
build should succeed

**Impact**
Annoyance.

**Logs and console output**
```
In file included from /workdir/zephyr/subsys/net/ip/net_context.c:40: 
/workdir/zephyr/subsys/net/ip/tcp2.h:120:1: error: useless type name in empty declaration [-Werror]
 void net_tcp_init(void); 

```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: (Zephyr SDK
 - Latest master


**Additional context**
It showed up in CI through PR #24873

--- a/subsys/net/ip/tcp2.h
+++ b/subsys/net/ip/tcp2.h
@@ -117,7 +117,11 @@ struct net_tcp_hdr *net_tcp_input(struct net_pkt *pkt,
 
 /* No ops, provided for compatibility with the old TCP */
 
+#if defined(CONFIG_NET_NATIVE_TCP)
 void net_tcp_init(void);
+#else
+#define net_tcp_init(...)
+#endif
 int net_tcp_update_recv_wnd(struct net_context *context, int32_t delta);
 int net_tcp_queue_data(struct net_context *context, struct net_pkt *pkt);
 int net_tcp_finalize(struct net_pkt *pkt);

